### PR TITLE
Build CUDA host code if CUDA libraries are available

### PR DESCRIPTION
**Pull request by fwyzard**
#### PR description:

Relax the condition for building CUDA host code to check only for the availability of the CUDA libraries, without requiring the possibility to compile CUDA device code.
This makes them available - for example - in GCC 10 builds.

#### PR validation:

The affected packages now build with GCC 10, and the `CUDAService` can be loaded:
```bash
$ echo $SCRAM_ARCH 
slc7_amd64_gcc820

$ cmsRun HeterogeneousCore/CUDAServices/test/testCUDAService.py 
%MSG-w CUDAService:  (NoModuleName) 02-Sep-2020 16:25:51 CEST pre-events
Failed to initialize the CUDA runtime.
Disabling the CUDAService.
%MSG
```

### CODE REVIEW

**Comment by fwyzard** [comment]
@cmsbuild, please test

**Comment by fwyzard** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
test parameters: 
  - release = slc7_amd64_gcc10

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
@smuzaffar Could you take a look that this makes sense?

(obviously this approach fails on architectures without any CUDA support, but I suppose we're not going to extend beyond amd64/aarch64/ppc64le (whatever that could be) anytime soon)

File: CUDADataFormats/BeamSpot/BuildFile.xml [hunk 1]
```diff
@@ -1,8 +1,10 @@
-<use name="rootcore"/>
-<use name="CUDADataFormats/Common"/>
-<use name="DataFormats/Common"/>
-<use name="HeterogeneousCore/CUDAUtilities"/>
+<iftool name="cuda"/>
```
Comment (smuzaffar) [inline CUDADataFormats/BeamSpot/BuildFile.xml:1]: @fwyzard , please remove the `/` from here

**Comment by smuzaffar** [comment]
@makortel looks good ( after this https://github.com/cms-sw/cmssw/pull/31334#discussion_r482707231  change ). 
These are correct changes. `<iftool name="cuda-gcc-support">` should be used only if we have some device code (`.cu` files) which we need to compile with `nvcc` otherwise `<iftool name="cuda">` should be enough.

> @smuzaffar Could you take a look that this makes sense?
> 
> (obviously this approach fails on architectures without any CUDA support, but I suppose we're not going to extend beyond amd64/aarch64/ppc64le (whatever that could be) anytime soon)

File: CUDADataFormats/BeamSpot/BuildFile.xml (hunk 1 above)
Comment (fwyzard) [reply CUDADataFormats/BeamSpot/BuildFile.xml]: done, thank you

**Comment by fwyzard** [comment]
@cmsbuild, please test

**Comment by fwyzard** [comment]
+1

**Comment by makortel** [comment]
test parameters:
  - release = slc7_amd64_gcc820

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by perrotta** [comment]
+1
- Technical, it does its job: it adds protection when cuda libraries are not available
- Jenkins tests pass both in "standard" architectures and witg gcc10

**Comment by christopheralanwest** [comment]
+alca

**Pull request merged**